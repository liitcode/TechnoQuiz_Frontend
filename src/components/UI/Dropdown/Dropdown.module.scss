
.selectBox {
  width: 100%;
  position: relative;
}

.selectBox:after {
  content: '';
  position: absolute;
  right: 10px;
  top: 50%;
  margin-top: -4px;
  border-top: 8px solid green;
  border-left: 6px solid transparent;
  pointer-events: none;
  border-right: 6px solid transparent;
}

.CategoryDropdown__select {
  width: 100%;
  height: 42px;
  -webkit-appearance: none;
  appearance: none;
  outline: 0;
  color: black;
  cursor: pointer;
  border: 2px solid #ccc;
  border-radius: 8px;
  line-height: 18px;
  padding: 10px 15px;
  font-size: 1rem;
  text-align-last: center;
}

.CategoryDropdown__select:focus {
  border: 2px solid green;
}

.CategoryDropdown__select option {
  height: 43px;
}

// option:hover {
//   background-color: yellow;
// }

select:hover {
  color: #444645;
  background: #ddd;
}

select option {
  margin: 40px;
  background-color: black;
  color: #fff;
  text-shadow: 0 1px 0 rgba(0, 0, 0, 0.4);
}
