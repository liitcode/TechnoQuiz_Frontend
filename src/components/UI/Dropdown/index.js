/* eslint-disable spaced-comment */

import React, { useState, useEffect } from 'react';
import axios from 'axios';
import styles from './Dropdown.module.scss';

function Dropdown() {
  const [data, setData] = useState([]);
  const [value, setValue] = useState('');
  useEffect(async () => {
    const result = await axios(
      'https://technoquiz-env.eba-33dpsiuk.ap-south-1.elasticbeanstalk.com/api/category',
    );
    setData(result.data);
  }, []);

  const optionSelected = (e) => {
    setValue(e.target.value);
  };

  return (
    <div className={styles.selectBox}>
      <select
        className={styles.CategoryDropdown__select}
        onChange={optionSelected}
        value={value}
      >
        <option value="">All Categories</option>
        {Object.keys(data).map((item) => (
          <option key={item} value={`${data[item]}`}>
            {data[item]}
          </option>
        ))}
      </select>
    </div>
  );
}

export default Dropdown;
