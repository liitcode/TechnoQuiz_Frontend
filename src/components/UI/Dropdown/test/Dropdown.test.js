import React from "react";
import renderer from "react-test-renderer";
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import storeConfig from '``../../../Redux/store/configureStore';
import Dropdown from "../index";

it('renders the component', () => {
    const tree = renderer.create(<Provider store={storeConfig}>
        <MemoryRouter><Dropdown /></MemoryRouter></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
})