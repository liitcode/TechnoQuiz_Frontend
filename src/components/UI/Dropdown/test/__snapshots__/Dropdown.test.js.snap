// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<div
  className="selectBox"
>
  <select
    className="CategoryDropdown__select"
    onChange={[Function]}
    value=""
  >
    <option
      value=""
    >
      All Categories
    </option>
  </select>
</div>
`;
